pkg/ip/link.go

```diff
@@ -66,27 +73,76 @@ func makeVeth(name string, mtu int) (peerName string, veth netlink.Link, err err
 	return
 }
 
-// RandomVethName returns string "veth" with random prefix (hashed from entropy)
-func RandomVethName() (string, error) {
-	entropy := make([]byte, 4)
+const ENTROPY_LEN = 4
+
+// RandomIfaceName returns string @prefix with random suffix (hashed from entropy)
+func RandomIfaceName(prefix string) (string, error) {
+	if len(prefix)+ENTROPY_LEN > syscall.IFNAMSIZ {
+		return "", fmt.Errorf("prefix %s too long", len(prefix))
+	}
+	entropy := make([]byte, ENTROPY_LEN)
 	_, err := rand.Reader.Read(entropy)
 	if err != nil {
 		return "", fmt.Errorf("failed to generate random veth name: %v", err)
 	}
 
-	// NetworkManager (recent versions) will ignore veth devices that start with "veth"
-	return fmt.Sprintf("veth%x", entropy), nil
+	// NetworkManager (1.0+) will ignore externally created virtual interfaces
+	return fmt.Sprintf("%s%x", prefix, entropy), nil
+}
+
+func RenameLink(curName, newName, newPrefix string) (string, error) {
+	if len(newName) == 0 && len(newPrefix) == 0 {
+		return "", fmt.Errorf("one of new name or new name prefix required")
+	}
+
+	link, err := netlink.LinkByName(curName)
+	if err != nil {
+		return "", err
+	}
+
+	if len(newName) > 0 {
+		if err := netlink.LinkSetName(link, newName); err != nil {
+			return "", err
+		}
+		return newName, nil
+	}
+
+	for idx := 0; idx < 100; idx++ {
+		newName = fmt.Sprintf(newPrefix, idx)
+		err = netlink.LinkSetName(link, newName)
+		if err == nil {
+			return newName, nil
+		} else if !os.IsExist(err) {
+			return "", err
+		}
+	}
+	return "", err
```

Comment:
Correction, it will be an `IsExist()`-type error. But the earlier mentioned magic 100 makes this weird. However, the error should indicate that the netns ran out of interfaces with an explicit message.